src/algebra/category/Module/algebra.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Module.monoidal
+import algebra.algebra.restrict_scalars
+import category_theory.linear
+
+/-!
+# Additional typeclass for modules over an algebra
+
+For an object in `M : Module A`, where `A` is a `k`-algebra,
+we provide additional typeclasses on the underlying type `M`,
+namely `module k M` and `is_scalar_tower k A M`.
+
+We provide the `linear k (Module A)` instance.
+
+## Note
+
+If you begin with a `[module k M] [module A M] [is_scalar_tower k A M]`,
+and build a bundled module via `Module.of A M`,
+these instances will not necessarily agree with the original ones.
+
+It seems without making a parallel version `Module' k A`, for modules over a `k`-algebra `A`,
+that carries these typeclasses, this seems hard to achieve.
+(An alternative would be to always require these typeclasses,
+requiring users to write `Module' ℤ A` when `A` is merely a ring.)
+-/
+
+universes v u
+open category_theory
+
+namespace Module
+
+variables {k : Type u} [field k]
+variables {A : Type u} [ring A] [algebra k A]
+
+instance (M : Module.{v} A) : module k M :=
+restrict_scalars.module k A M
+
+instance (M : Module.{v} A) : is_scalar_tower k A M :=
+restrict_scalars.is_scalar_tower k A M
```

Comment:
What do you think of making these `def`s? And then making the local instances when needed? (Maybe with `open_locale Module` or something like that?)

Most of the time, when working with a few objects, it is probably better to assume `[module k M]` and `[is_scalar_tower k A M]`. But of course that doesn't work to get `linear k (Module.{v} A)`.

Reply:
I'm not sure what this is to worry about with these as instances. They are "structural", so will only ever fire if you actually have the coercion of a `Module` to `Type` present.

Reply:
It might be helpful to look at #14023, particularly [here](https://github.com/leanprover-community/mathlib/pull/14023/files#diff-36baaab5db4ee9560da3bcedde9a281da50b711e8a51e18352498803ac0e573cR38), to see how I'm intending to use these instances.

Reply:
Okay, I'll see if I can cope without these as instances.